### PR TITLE
front: fix the warning display above the GET

### LINKED ISSUE #8878: front: the warning above the GET doesn't disappear if a timetable has some invalid trains
### What happened?

When the GET is loading, a warning is displayed above the GET to inform the user that the data is loading.
However, if the timetable has some invalid trains, then the warning never disappear.
image

### What did you expect to happen?

The warning should disappear once all the trains are projected

### How can we reproduce it (as minimally and precisely as possible)?

1. Open a scenario with some valid and invalid trains
2. Select a valid train and project on a valid train
3. See the GET

### On which environments the bug occurs?

_No response_

### On which browser the bug occurs?

_No response_

### OSRD version (top right corner `Account` button > `Informations`)

.

### PR DESCRIPTION
**Pull request by clarani**
closes #8878 

Add a new boolean allTrainsProjected and display the warning "Projecting..." only if this boolean is not truthy.

### CODE REVIEW

**Comment by emersion** [comment]
Could we instead use a `useMemo` with a callback checking `shouldProjectIds.length === 0`?

**Comment by anisometropie** [comment]
it doesn’t go away after I just created 100 trains

Screenshot_20240918_134915

**Comment by anisometropie** [comment]
weirdly the too figures are not synchronized 40 on the left 30 on the right
Screenshot_20240918_135020

**Comment by clarani** [comment]
@anisometropie thanks for your review :pray: 
- thanks for the bug, I fixed it in the 3rd commit
- for your 2nd remark, actually it's normal that the counters are de-synchronized, since the loading of the trains and the projection are done in parallel and projection is done after the loading

**Comment by emersion** [comment]
Another way to fix this without a desync would be to subtract the number of invalid trains to the total number of simulated trains. But the current approach here looks fine to me as well.

**Review by anisometropie — APPROVED** [review]
thanks!

File: front/src/modules/simulationResult/components/SpaceTimeChart/useLazyProjectTrains.ts [hunk 1]
```diff
@@ -121,7 +122,7 @@ const useLazyProjectTrains = ({
     if (
       !moreTrainsToCome &&
       trainIdsToProject.length > 0 &&
-      requestedProjectedTrainIds.current.size === trainIdsToProject.length
+      requestedProjectedTrainIds.current.size === removeDuplicates(trainIdsToProject).length
```
Comment (emersion) [inline front/src/modules/simulationResult/components/SpaceTimeChart/useLazyProjectTrains.ts]: Would be nicer to use a `Set` for `trainIdsToProject`.

**Review by emersion — APPROVED** [review]
LGTM regardless

**Review by kmer2016 — APPROVED** [review]
LGTM

**Pull request merged**